File: Algorithm_Java/src/Stack/stack1.java
package Stack;
import java.util.Scanner;
import java.util.Stack;

public class stack1 {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int k = sc.nextInt();
        Stack<Integer> s = new Stack<>();

        for(int i = 0; i < k; i++){
            String str = sc.next();
            if(str.equals("push")){
                int n = sc.nextInt();
                s.push(n);
            }
            else if(str.equals("pop")){
                System.out.println(s.peek());
                s.pop();
            }
            else if(str.equals("size")){
                System.out.println(s.size());
            }
            else if(str.equals("empty")){
                if(s.isEmpty()){
                    System.out.println("1");
                }
                else System.out.println("0");
            }
            else if(str.equals("top")){
                System.out.println(s.peek());

            }
        }
    }
}

File: Algorithm_Cpp/Algorithm_Cpp/Stack/stack1.cpp
#include <iostream>
#include <string>
#include <stack>

using namespace std;

int stack1() {

    int k;
    cin >> k;
    stack<int> s;
    string str;

    for(int i = 0; i < k; i++){
        cin >> str;
        if(str == "push"){
            int n;
            cin >> n;
            s.push(n);
        }
        else if(str == "pop"){
            cout << s.top() << endl;
            s.pop();
        }
        else if(str == "size"){
            cout << s.size() << endl;
        }
        else if(str == "empty"){
            cout << s.empty() << endl;
        }
        else if(str == "top"){
            cout << s.top() << endl;
            
        }
    }

    return 0;
}
